workbench/src/main/settingsStore.js

```diff
@@ -0,0 +1,88 @@
+import { app, ipcMain } from 'electron';
+import Store from 'electron-store';
+import Ajv from 'ajv';
+
+import { ipcMainChannels } from './ipcMainChannels';
+import { getLogger } from './logger';
+
+const logger = getLogger(__filename.split('/').slice(-1)[0]);
+
+export const defaults = {
+  nWorkers: '-1',
+  taskgraphLoggingLevel: 'INFO',
+  loggingLevel: 'INFO',
+  language: 'en',
+};
+
+export const schema = {
+  type: 'object',
+  properties: {
+    nWorkers: {
+      type: 'string',
+    },
+    taskgraphLoggingLevel: {
+      enum: ['CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG'],
+    },
+    loggingLevel: {
+      enum: ['CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG'],
+    },
+    language: {
+      enum: ['en', 'es', 'zh'],
```

Comment:
That's an interesting question. Normally I would say, yes, but after some thought I feel like it's safest to keep these hardcoded. If any unexpected values ever made it into this schema, then unexpected things might happen, and the whole app might crash. I guess we could be prepared to catch that, but I just feel more at ease by making that fail case impossible by using a statically defined schema rather than a dynamic one.